hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/ozone/om/TestAddRemoveOzoneManager.java

```diff
@@ -333,4 +340,78 @@ public void testForceBootstrap() throws Exception {
     // Verify that the newly bootstrapped OM is running
     Assert.assertTrue(newOM.isRunning());
   }
+
+  /**
+   * Decommissioning Tests:
+   * 1. Stop an OM and decommission it from a 3 node cluster
+   * 2. Decommission another OM without stopping it.
+   * 3.
+   */
+  @Test
+  public void testDecommission() throws Exception {
+    setupCluster(3);
+    user = UserGroupInformation.getCurrentUser();
+
+    // Stop the 3rd OM and decommission it
+    String omNodeId3 = cluster.getOzoneManager(2).getOMNodeId();
+    cluster.stopOzoneManager(omNodeId3);
+    decommissionOM(omNodeId3);
+
+    // Decommission an OM and then stop it. Stopping OM before will lead
+    // to no quorum and there will not be a elected leader OM to process the
+    // decommission request.
+    String omNodeId2 = cluster.getOzoneManager(1).getOMNodeId();
+    decommissionOM(omNodeId2);
+    cluster.stopOzoneManager(omNodeId2);
+
+    // Verify that we can read/ write to the cluster with only 1 OM.
+    OzoneVolume volume = objectStore.getVolume(VOLUME_NAME);
```

Comment:
From above test we stopped 2 OM's how read/write will succeed?

Reply:
Once we remove 2 nodes, the ratis ring becomes a single node ring.

Reply:
Got it. Now it makes sense why test passes :)